### PR TITLE
Validated expected event numbers for damoncontroller test.

### LINKED ISSUE #46629: Validating expected Event numbers for damoncontroller test
**Is this a request for help?** (If yes, you should use our troubleshooting guide and community support channels, see https://kubernetes.io/docs/tasks/debug-application-cluster/troubleshooting/.):

**Note:** Please file issues for subcomponents under the appropriate repo

| Component | Repo                                                               |
| --------- | ------------------------------------------------------------------ |
| kubectl   | [kubernetes/kubectl](https://github.com/kubernetes/kubectl/issues/new) |
| kubeadm   | [kubernetes/kubeadm](https://github.com/kubernetes/kubeadm/issues/new) |

**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.):

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

Now the [daemoncontrol test  do not include verify for daemonset event number](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/daemon/daemoncontroller_test.go#L274), we should add this check in the unit test.

such as 
```
func validateSyncDaemonSets(t *testing.T, fakePodControl *fakePodControl, expectedCreates, expectedDeletes int, expectedEvents int) {
```

/cc @wojtek-t

### PR DESCRIPTION
**Pull request by gyliu513**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #46629

**Special notes for your reviewer**:

**Release note**:

```release-note
none
```

### CODE REVIEW

**Comment by gyliu513** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by gyliu513** [comment]
@lukaszo @janetkuo ping

**Review by lukaszo — APPROVED** [review]

**Comment by lukaszo** [comment]
/approve

Please note that it will wait for 1.8

File: pkg/controller/daemon/daemoncontroller_test.go [hunk 1]
```diff
@@ -523,7 +527,13 @@ func TestInsufficientCapacityNodeDaemonDoesNotLaunchPod(t *testing.T) {
 			Spec: podSpec,
 		})
 		manager.dsStore.Add(ds)
-		syncAndValidateDaemonSets(t, manager, ds, podControl, 0, 0)
+		if apiequality.Semantic.DeepEqual(strategy, newOnDeleteStrategy()) {
```
Comment (janetkuo) [inline pkg/controller/daemon/daemoncontroller_test.go]:
```go 
switch strategy.Type {
case extensions.OnDeleteDaemonSetStrategyType:
...
}
```

Comment (gyliu513) [inline pkg/controller/daemon/daemoncontroller_test.go]: Done

**Comment by gyliu513** [comment]
@janetkuo @lukaszo can you please help approve this?

**Comment by gyliu513** [comment]
Ping @janetkuo @lukaszo I think that this is ready to go, thanks.

**Comment by gyliu513** [comment]
ping @mikedanese @janetkuo for approve

**Pull request merged**